src/module_PI/Raspberry_PI/JUnitTests/SerialInputListenerTest.java

```diff
@@ -4,31 +4,34 @@
 
 import org.junit.Test;
 
-public class SerialInputListenerTest {
-
-	@Test
-	public void testRun() {
-		fail("Not yet implemented"); // TODO
-	}
+import module_PI.Raspberry_PI.main.SerialInputListener;
 
-	@Test
-	public void testSerialInputListener() {
-		fail("Not yet implemented"); // TODO
-	}
+public class SerialInputListenerTest {
 
 	@Test
 	public void testIsPacketAvailable() {
-		fail("Not yet implemented"); // TODO
+		SerialInputListener inputListener = new SerialInputListener();
+		assertFalse(inputListener.isPacketAvailable());
 	}
 
 	@Test
 	public void testIsDataStreamEmpty() {
-		fail("Not yet implemented"); // TODO
+		SerialInputListener inputListener = new SerialInputListener();
+		assertTrue(inputListener.isDataStreamEmpty());
 	}
 
+/*TODO Have Arduino Respond To Ping Request, then write this test.
```

Comment:
I noticed this comment. I'm not suggesting any changes at this moment but just some ideas to consider when you come back to this comment/test:

For "_Unit Testing_" you don't want to rely on an outside dependency like this. You wouldn't want to have the actual Arduino respond to a ping.

For "_Acceptance Testing_" (_or possibly "Integration Testing" or "System Testing"_) - http://softwaretestingfundamentals.com/acceptance-testing/ you want want to test the 'interaction ' or 'integration' with external dependencies. This helps check to make sure the 'contract' between the two things is still correct. Those tests are generally written in a test file of their own and run separately from Unit Tests.

To accomplish the test you look like you are wanting to do here there are multiple 'mocking' frameworks out there that will let you accomplish basically what you're wanting to do. In essence it is like making a "fake arduino" that you'd "call" and get a packet response back from. 

One such mocking framework is [Mockito](http://site.mockito.org/) (_another I've heard a bunch about is called [EasyMock](http://easymock.org) though I've never used it myself_). With it you'd end up doing something similar to declaring a mock for the sender and then do a line something like ```when(mockedSerialSender.isPacketAvailable()).thenReturn(true);``` (Obviously I haven't looked in detail at what the actual calls would be but thats just an example). Then you would make the call and move forward with whatever test you would want to accomplish.

So thats just some things to chew on and think about in terms of testing. Its quite common to mock things like this in larger projects where there are multiple external dependencies.